server/src/routes/donatedItemRoutes.ts

```diff
@@ -145,43 +145,74 @@ router.get('/:id', async (req: Request, res: Response) => {
     }
 });
 
-// PUT /donatedItem/details/:id - Update non-status details of a DonatedItem
+// PUT /donatedItem/status/:id - Update the status of a DonatedItem and handle image uploads
 router.put(
-    '/details/:id',
-    donatedItemValidator,
+    '/status/:id',
+    upload.array('imageFiles'), // Handle image files as multipart
     async (req: Request, res: Response) => {
         try {
-            const donorId = parseInt(req.body.donorId);
-            const programId = parseInt(req.body.programId);
-            try {
-                await validateDonor(donorId);
-                await validateProgram(programId);
-            } catch (error) {
-                if (error instanceof Error) {
-                    return res.status(400).json({ error: error.message });
-                }
+            const donatedItemId = parseInt(req.params.id);
+
+            // Validate donated item ID
+            await validateDonatedItem(donatedItemId);
+
+            // Extract status details and image files
+            const { statusType, dateModified } = req.body;
+            const imageFiles = req.files as Express.Multer.File[];
+
+            // Ensure statusType and dateModified are provided
+            if (!statusType || !dateModified) {
+                return res
+                    .status(400)
+                    .json({ error: 'statusType and dateModified are required' });
```

Comment:
Enhance validation for statusType and dateModified to check for valid formats or values.